### PR TITLE
Add parseFromByteArray

### PR DESCRIPTION
**Pull request by travisbrown**
We've talked about wanting this at Permutive and it seems like the right time to add it. It's a pretty simple copy-paste of `ByteBufferParser`.

### CODE REVIEW

File: parser/src/main/scala/jawn/ByteArrayParser.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package org.typelevel.jawn
+
+import scala.annotation.{switch, tailrec}
+import java.nio.{ByteBuffer, Buffer}
```
Comment (larsrh) [inline parser/src/main/scala/jawn/ByteArrayParser.scala]: Unneeded import?

Comment (travisbrown) [reply parser/src/main/scala/jawn/ByteArrayParser.scala]: Thanks, good catch—fixing now.

**Comment by travisbrown** [comment]
@larsrh I went ahead and checked `-Xlint` to make sure there weren't other unused imports, and turned up a few, which I've gone ahead and fixed here. We can't enable it for the build because we have fatal warnings and there's one warning I didn't fix (an unused implicit that I'll address in a follow-up PR).

File: parser/src/main/scala/jawn/ByteArrayParser.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package org.typelevel.jawn
+
+/**
+ * Basic byte array parser.
+ */
+final class ByteArrayParser[J](src: Array[Byte]) extends SyncParser[J] with ByteBasedParser[J] {
```
Comment (rossabaker) [inline parser/src/main/scala/jawn/ByteArrayParser.scala]: Some of these are `private[jawn]` (e.g., `StringParser`), and some are public.  Is there a reason any of them are public?  Something to think about before 1.0.

Comment (travisbrown) [reply parser/src/main/scala/jawn/ByteArrayParser.scala]: I noticed this today and thought I could remember noticing it before and finding out there was some reason for it. I'll look again before 1.0.0.

**Pull request merged**